### PR TITLE
Issue/353/numcosmo examples

### PR DESCRIPTION
**Pull request by pennalima**
Examples and documentation updated #353

### CODE REVIEW

**Review by m-aguena — CHANGES_REQUESTED** [review]
Please check nbs again with master merged into it

**Comment by pennalima** [comment]
Hi @m-aguena , both examples are running properly. 
 I am just modifying the last figure, the current x and y ranges are not good (the contour and one distribution are cut).

**Comment by pennalima** [comment]
Done!

**Review by combet — CHANGES_REQUESTED** [review]
Hi @pennalima and @vitenti! Thanks for the updated version of the NB. 

Here are a couple of things I'd like your input on:
- update the opening text at the very beginning of the two notebooks, as it still mentions the use of `scipy.optimize` in both notebooks. I think it would be good to also mention in the opening statements that these NB mirror the 'default' ones, but make use of NC's statistical framework for the parameter estimations. And that you're also letting the concentration as a free parameter in Example 2, which is not the case in the 'default' notebooks.
- I don't understand in cell 18 of Example2 why you need to set e.g., `moo1.set_concentration (4.0)`, given that you leave the concentration to be a free parameter. I understand it in example1, where the output says ` cDelta[00]:  4                   [FIXED]`, but not for Example2
- I'm wondering whether we should have a specific Numcosmo example folder to store all NC notebooks?

**Comment by pennalima** [comment]
Done! @combet

**Comment by m-aguena** [comment]
@pennalima I tried to run the notebook and got:
```
AttributeError                            Traceback (most recent call last)
<ipython-input-23-a61c50eff22f> in do_mean_func(self, mset, vp)
     38 
     39     def do_mean_func (self, mset, vp):
---> 40         vp.set_array (self.moo.eval_reduced_shear (self.props.r_source.dup_array (), self.props.z_cluster, self.props.z_source.dup_array ()))
     41         return
     42 

AttributeError: 'NumCosmoCLMModeling' object has no attribute 'eval_reduced_shear'
```

I believe this function is called `eval_reduced_tangential_shear` now.

**Review by combet — APPROVED** [review]
Thanks @pennalima - all seems good now. If you get the chance, could you please delete the NB outputs, otherwise I'll take care of it before merging.

**Comment by combet** [comment]
> @pennalima I tried to run the notebook and got:
> 
> ```
> AttributeError                            Traceback (most recent call last)
> <ipython-input-23-a61c50eff22f> in do_mean_func(self, mset, vp)
>      38 
>      39     def do_mean_func (self, mset, vp):
> ---> 40         vp.set_array (self.moo.eval_reduced_shear (self.props.r_source.dup_array (), self.props.z_cluster, self.props.z_source.dup_array ()))
>      41         return
>      42 
> 
> AttributeError: 'NumCosmoCLMModeling' object has no attribute 'eval_reduced_shear'
> ```
> 
> I believe this function is called `eval_reduced_tangential_shear` now.



> @pennalima I tried to run the notebook and got:
> 
> ```
> AttributeError                            Traceback (most recent call last)
> <ipython-input-23-a61c50eff22f> in do_mean_func(self, mset, vp)
>      38 
>      39     def do_mean_func (self, mset, vp):
> ---> 40         vp.set_array (self.moo.eval_reduced_shear (self.props.r_source.dup_array (), self.props.z_cluster, self.props.z_source.dup_array ()))
>      41         return
>      42 
> 
> AttributeError: 'NumCosmoCLMModeling' object has no attribute 'eval_reduced_shear'
> ```
> 
> I believe this function is called `eval_reduced_tangential_shear` now.

I'm very puzzled - it was running fine...

**Comment by combet** [comment]
> > @pennalima I tried to run the notebook and got:
> > ```
> > AttributeError                            Traceback (most recent call last)
> > <ipython-input-23-a61c50eff22f> in do_mean_func(self, mset, vp)
> >      38 
> >      39     def do_mean_func (self, mset, vp):
> > ---> 40         vp.set_array (self.moo.eval_reduced_shear (self.props.r_source.dup_array (), self.props.z_cluster, self.props.z_source.dup_array ()))
> >      41         return
> >      42 
> > 
> > AttributeError: 'NumCosmoCLMModeling' object has no attribute 'eval_reduced_shear'
> > ```
> > 
> > 
> > I believe this function is called `eval_reduced_tangential_shear` now.
> 
> I'm very puzzled - it was running fine...

OK, I hadn't activate the right conda env, so I was still running with v0.8.0 where the functions hadn't changed name - so it was working fine. Indeed, in both notebooks, we need `eval_reduced_tangential_shear`. Good you caught that @m-aguena !

**Comment by combet** [comment]
@m-aguena, could you  please check all works for you now (I pushed the changes, but I'm online with Mariana and we looked at it together).

File: examples/NumCosmo/Example2_Fit_Halo_Mass_to_Shear_Catalog_NC.ipynb [hunk 1]
```diff
@@ -0,0 +1,944 @@
...
+    "import gi\n",
+    "\n",
+    "gi.require_version('NumCosmo', '1.0')\n",
+    "gi.require_version('NumCosmoMath', '1.0')\n",
+    "from gi.repository import GObject\n",
+    "from gi.repository import NumCosmo as Nc\n",
+    "from gi.repository import NumCosmoMath as Ncm\n",
+    "\n",
+    "from scipy.stats import chi2\n",
+    "import corner\n",
```
Comment (m-aguena) [inline examples/NumCosmo/Example2_Fit_Halo_Mass_to_Shear_Catalog_NC.ipynb:56]: This package is not installed in our default instructions, maybe add a comment here to tell the user install it if he does not have it

Comment (combet) [reply examples/NumCosmo/Example2_Fit_Halo_Mass_to_Shear_Catalog_NC.ipynb]: Added a comment in both the introduction text and just above the import.

**Comment by combet** [comment]
Hi @m-aguena, do you consider this PR OK now? Let me know and I'll go ahead and merge. Thanks

**Comment by m-aguena** [comment]
@combet Yes, go ahead

**Pull request merged**